src/rules/enforce-memoize-async.ts

```
@@ -57,32 +58,77 @@ export const enforceMemoizeAsync = createRule<Options, MessageIds>({
           return;
         }
 
-        // Check if method already has @Memoize decorator
+        // Check if method already has @Memoize or @Memoize() decorator
         const hasDecorator = node.decorators?.some((decorator) => {
-          if (decorator.expression.type !== AST_NODE_TYPES.CallExpression) {
-            return false;
+          const expression = decorator.expression as any;
```

ESLint

[error] 'expression' is assigned a value but never used.

(@typescript-eslint/no-unused-vars)